data_steward/cdr_cleaner/cleaning_rules/remove_invalid_procedure_source_records.py

```diff
@@ -38,12 +43,20 @@
  SELECT
     concept_id
   FROM
-    `{DATASET_ID}.concept`
+    `{project}.{dataset}.concept`
   WHERE
     TRIM(concept_class_id) = 'CPT4 Modifier'
 )
 """
 
+VALID_PROCEDURE_SOURCE_CONCEPT_IDS_QUERY = """
+SELECT * FROM
+`{project}.{dataset}.procedure_occurrence`
+WHERE
+procedure_occurrence_id
+IN (SELECT procedure_occurrence_id FROM `{project}.{sandbox_dataset}.{intermediary_table}`)
+"""
```

Comment:
Another question.  Based on the variable name `INVALID_PROCEDURE_SOURCE_CONCEPT_IDS_QUERY`, it seems like the sandbox should be storing the records we don't want to see in the cleaned table.  If that assumption is correct, should line 57 be using a `NOT EXISTS` clause in place of an `IN` clause?

Reply:
yep, 100%. this is my bad. changing to NOT EXISTS

Reply:
@lrwb-aou this is fixed, but please double check `NOT EXISTS` this is my first time using, so I'm not sure if it's correct. I would have typically used the following, but as we discussed previously, I think it would exclude all null values as well. 

```
SELECT * FROM
`{project}.{dataset}.procedure_occurrence`
WHERE
procedure_occurrence_id
NOT IN (SELECT procedure_occurrence_id FROM `{project}.{sandbox_dataset}.{intermediary_table}`)
```

Reply:
Unfortunately, this not exists does not work.  I tested by putting three rows in a sandbox table and running the query created to select only the rows you want.  It did not return any results.  I expected 63973754 records.  You can view my test dataset, `lrwb_deid_test_input_20190910`  and its now associated sandbox `lrwb_deid_test_input_20190910_sandbox`.  

This is a working `NOT EXISTS` query using the sandbox dataset.

```
select * from `aou-res-curation-test.lrwb_deid_test_input_20190910.procedure_occurrence` as p0
where not exists (
select p1.procedure_occurrence_id FROM `aou-res-curation-test.lrwb_deid_test_input_20190910_sandbox.procedure_occurrence_dc583` as p1 where p0.procedure_occurrence_id = p1.procedure_occurrence_id)
```
This did return 63973754 results.  There are 63973757 records in the input procedure_occurrence table.  There are 3 records in the sandbox table.